src/CommunityToolkit.Maui.Core/Interfaces/IAvatarView.shared.cs

```diff
@@ -0,0 +1,73 @@
+﻿namespace CommunityToolkit.Maui.Core;
+
+/// <summary>Avatar view interface.</summary>
+public interface IAvatarView : IBorderView, ILabel, Microsoft.Maui.IImage, IImageSource
```

Comment:
Interfaces are intended to be used for controls that will implement a `Handler`, which isn't the case of the `AvatarView` so we don't need the interface here, and since this control uses the `Maui.View` we don't need to have anything on `Core`, this can be all `Maui` specific.

Also, this interface has a lot of MVVM (the propertyChanged methods) smell and the `Core` project must not influence the implementor in any way. One more reason that we don't need it.

Reply:
> Interfaces are intended to be used for controls that will implement a `Handler`, which isn't the case of the `AvatarView` so we don't need the interface here, and since this control uses the `Maui.View` we don't need to have anything on `Core`, this can be all `Maui` specific.
> 
> Also, this interface has a lot of MVVM (the propertyChanged methods) smell and the `Core` project must not influence the implementor in any way. One more reason that we don't need it.

@pictos I was asked to move IAvatarView to `Core` in [This commen](https://github.com/CommunityToolkit/Maui/pull/481#pullrequestreview-1029242225).  Appreciate that the code has evolved, but I'm getting conflicting requests.  I'll change as to what is best, but not sure right now.

Reply:
We don't need to have an interface at all, as I said, interfaces are needed if the control will implement a `Handler`, the `AvatarView` is just a composition of Maui controls

Reply:
The interface isn't necessarily a bad thing, because it'll help future toolkits like `CommunityToolkit.Comet` and `CommunityToolkit.BlazorWebView` implement their own AvatarView. 

But, I agree that it's a good idea to ensure it's not MVVM-specific. [Comet](https://github.com/dotnet/comet), for example, uses the MVU architecture. 

Does [Comet](https://github.com/dotnet/comet), for example, fire events like `OnCornerRadiusPropertyChanged()` and does it use methods like `IsTextSet()`? Or is that just an MVVM thing?

If other MAUI-based frameworks like Comet and [BlazorWebView](https://github.com/dotnet/maui/tree/main/src/BlazorWebView) don't leverage these method calls, then we should remove them from the interface.

Reply:
> The interface isn't necessarily a bad thing, because it'll help future toolkits like CommunityToolkit.Comet and CommunityToolkit.BlazorWebView implement their own AvatarView. (These future toolkits will also have a dependency on CommunityToolkit.Maui.Core).

I can see your point, and I'm on the fence. But if we keep the interface we also should avoid things like `Notes to implementer`, since we don't know where this will be implemented we shouldn't be restrictive. And this is why I don't like to use the interface in this case, we don't know which features other targets will provide, so this interface will be a good guide or a restrictive guide on how this `View` should be implemented?

> Does [Comet](https://github.com/dotnet/comet), for example, fire events like OnCornerRadiusPropertyChanged() and does it use methods like IsTextSet()? Or is that just an MVVM thing?

The `Changed` method is an MVVM  thing, about the `TextSet` I'm not sure but I would say it's an MVVM thing, MVU just `redraw` the UI. The interface doesn't need those `Changed` methods. 
Makes sense to have methods that describe an action (not the .NET one), for example, the [`IButton`](https://github.com/dotnet/maui/blob/main/src/Core/src/Core/IButton.cs) you have just methods that describe the `Pressed`, `Released`, and `Clicked` actions.

Reply:
Where I was going, for good or ill, I was trying to mirror as much from `Microsoft.Maui.Controls` as possible.  With a goal of making the control as easy to use for the end user as possible in it's simplest form, but still give the end user as much extensability as possible/feasible.

As such I took things like `Microsoft.Maui.Controls.ImageButton` as an example, and used this as my guide.  `ImageButton` inherits the interface `Microsoft.Maui.Controls.Internals.IButtonElement`, which has a seperation between properties and methods and includes **note to implementor**.  As such, I copied that, with the thought of "If it's included by Microsoft, then why shouldn't I?"

I do however totally get what @pictos is saying, in that including the note could be regards as restrictive, especially if this `view` is used as a guide going forward. Does the toolkit use `Microsoft.Maui` as a guide, or does it set it own?

Reply:
I do not have objections to add interface. But in that case we also need t add it for other controls like UniformItemsLayout

Reply:
@GeorgeLeithead, I know that you followed the Maui structure, but there're some details (that we should probably document). Bear with me for a while.

If you see the [`IButton`](https://github.com/dotnet/maui/blob/main/src/Core/src/Core/IButton.cs) interface here, it's in the `Microsoft.Maui` namespace, and in the `Core` project, so it should be agnostic in terms of patterns. Now if you see the [`ITextAlignmentElement`](https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/ITextAlignmentElement.cs#L11) interface there's the `propertyChanged` method, but look! The namespace is `Microsoft.Maui.Controls` and it lives on the `Control` assembly, where the implementation isn't agnostic, in this case, is XAML and MVVM friendly.

On this project we followed this same idea, everything in the `Core` assembly should be agnostic in terms of implementation. And inside the `.Maui` assembly we are XAML and MVVM friendly.

I hope that makes things more clear to you, if don't feel free to asking questions, I'll be more than happy in provide all info that you need (:

Reply:
As everything I'm 'currently' doing IS XAML/MVVM friendly. Should I rename to `IAvatarViewElement` and move to `CommunityToolKit.Maui` namespace? (beside `AvatarView.shared.cs`). Then later if needed, Handler implementations could be added.

Reply:
There's no reason to have such an interface in the `.Maui` project. But I'll let @brminnick decide this, since he wants to have the `interface` in the shared layer to provide a guidance for other implementors.